src/common/download.h

```diff
@@ -32,5 +32,11 @@
 
 namespace tools
 {
-  bool download(const std::string &path, const std::string &url);
+  typedef void *download_async_handle;
```

Comment:
In fact, if type safety is what's wanted, I might just do:

  struct download_thread_control;
  typedef download_thread_control download_async_handle;

Or even use download_thread_control as is (maybe renaming it, as this was internal). Am I missing an advantage your suggestion has ?